src/Validator/Parser/hdtXSDSchemaParser.cpp

```
@@ -0,0 +1,413 @@
+				}
+				collectEnumerations(elem, vals);
+				if (!vals.empty())
+					result[name] = std::move(vals);
+			}
+
+			return result;
+		}
+
+		/// Parses required attributes per top-level element.
+		std::unordered_map<std::string, std::vector<std::string>>
+			parseAllRequiredAttrs(const pugi::xml_document& doc)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)